### PR TITLE
docs: fix reactive forms language and example

### PR DESCRIPTION
**Pull request by kapunahelewong**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #2209 and #21968

## What is the new behavior?
change to not call `ngOnChanges` directly, changes button from type `reset`, and reworks text to reflect these changes. Also clarifies error map info.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: aio/content/guide/reactive-forms.md [hunk 1]
```diff
@@ -263,167 +238,193 @@ Do the following two things in `app.module.ts`:
...
 Revise the `AppComponent` template so it displays the `HeroDetailComponent`.
 
 <code-example path="reactive-forms/src/app/app.component.1.html" title="src/app/app.component.html" linenums="false">
 
 </code-example>
 
+{@a essentials}
 
+## Essential form classes
+This guide uses four fundamental classes to build a reactive forms:
```
Comment (kara) [inline aio/content/guide/reactive-forms.md]: Typo: `a reactive forms` -> `a reactive form`

File: aio/content/guide/reactive-forms.md [hunk 2]
```diff
@@ -263,167 +238,193 @@ Do the following two things in `app.module.ts`:
...
+      <code>AbstractControl</code>
+    </td>
+
+    <td>
+
+      [`AbstractControl`](api/forms/AbstractControl "API Reference: FormControl") is the abstract base class for the three concrete form control classes;
 `FormControl`, `FormGroup`, and `FormArray`.
-It provides their common behaviors and properties, some of which are _observable_.
+It provides their common behaviors and properties,
+some of which are observable.
```
Comment (kara) [inline aio/content/guide/reactive-forms.md]: I'm not sure I understand `some of which are observable`. All of them are observable, no? Maybe it's supposed to be some of them are `observables`? I think it should be removed bc it's confusing.

File: aio/content/guide/reactive-forms.md [hunk 3]
```diff
@@ -263,167 +238,193 @@ Do the following two things in `app.module.ts`:
...
+    </td>
+
+    <td>
 
-* [_FormControl_](api/forms/FormControl "API Reference: FormControl")
-tracks the value and validity status of an _individual_ form control.
-It corresponds to an HTML form control such as an input box or selector.
+      [`FormControl`](api/forms/FormControl "API Reference: FormControl")
+tracks the value and validity status of an individual form control.
+It corresponds to an HTML form control such as an `<input>` or `<selector>`.
```
Comment (kara) [inline aio/content/guide/reactive-forms.md]: You mean `<select>` here, I think?

File: aio/content/examples/reactive-forms/src/app/hero-detail/hero-detail.component.ts [hunk 4]
```diff
@@ -42,6 +45,10 @@ export class HeroDetailComponent implements OnChanges {
   }
 
   ngOnChanges() {
+    this.rebuildForm();
+  }
+
+  rebuildForm() {
     this.heroForm.reset({
       name: this.hero.name
     });
```
Comment (kara) [inline aio/content/examples/reactive-forms/src/app/hero-detail/hero-detail.component.ts]: I can't comment directly on it, but looks like there is still an `ngOnChanges` reference in `onSubmit` below (line 76)

**Review by kara — COMMENTED** [review]
Small nits

File: aio/content/guide/reactive-forms.md [hunk 5]
```diff
@@ -106,15 +106,15 @@ You must wrap your test block in `async()` or `fakeAsync()` to
 avoid looking for values in the form that aren't there yet.
 With reactive forms, everything is available when you expect it to be.
 
-### Which is better, reactive or template-driven?
+### Choosing reactive or template-driven forms
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/guide/reactive-forms.md]: Thank you!

File: aio/content/guide/reactive-forms.md (hunk 1 above)
Comment (kapunahelewong) [reply aio/content/guide/reactive-forms.md]: Fixed!

File: aio/content/guide/reactive-forms.md (hunk 2 above)
Comment (kapunahelewong) [reply aio/content/guide/reactive-forms.md]: Agreed. Removed that phrase.

File: aio/content/guide/reactive-forms.md (hunk 3 above)
Comment (kapunahelewong) [reply aio/content/guide/reactive-forms.md]: Ha! Yes, thank you for catching that. Fixed. :)

File: aio/content/examples/reactive-forms/src/app/hero-detail/hero-detail.component.ts (hunk 4 above)
Comment (kapunahelewong) [reply aio/content/examples/reactive-forms/src/app/hero-detail/hero-detail.component.ts]: Ohhhh I see. Yes, changed to `rebuildForm()`.

**Review by kara — APPROVED** [review]
LGTM

**Pull request closed without merging**